tests/benchmarks/micro_benchmarks/test_disk_performance.py

```diff
@@ -0,0 +1,493 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Tests for disk-performance benchmark."""
+
+from pathlib import Path
+import os
+import unittest
+
+from superbench.benchmarks import BenchmarkRegistry, BenchmarkType, ReturnCode, Platform
+
+
+class DiskPerformanceTest(unittest.TestCase):
+    """Test class for disk-performance benchmark."""
+    def setUp(self):
+        """Method called to prepare the test fixture."""
+        # Create fake binary file just for testing.
+        os.environ['SB_MICRO_PATH'] = '/tmp/superbench/'
+        binary_path = os.path.join(os.getenv('SB_MICRO_PATH'), 'bin')
+        Path(binary_path).mkdir(parents=True, exist_ok=True)
+        self.__binary_file = Path(binary_path, 'fio')
+        self.__binary_file.touch(mode=0o755, exist_ok=True)
+
+    def tearDown(self):
+        """Method called after the test method has been called and the result recorded."""
+        self.__binary_file.unlink()
+
+    def test_disk_performance_command_generation(self):
+        """Test disk-performance benchmark command generation."""
+        benchmark_name = 'disk-performance'
+        (benchmark_class,
+         predefine_params) = BenchmarkRegistry._BenchmarkRegistry__select_benchmark(benchmark_name, Platform.CPU)
+        assert (benchmark_class)
+
+        # Test case 1: empty parameter
+        benchmark = benchmark_class(benchmark_name, parameters='')
+
+        # Check basic information
+        assert (benchmark)
+        ret = benchmark._preprocess()
+        assert (ret is True)
+        assert (benchmark.return_code == ReturnCode.SUCCESS)
+        assert (benchmark.name == 'disk-performance')
+        assert (benchmark.type == BenchmarkType.MICRO)
+
+        # Command list should be empty
+        assert (0 == len(benchmark._commands))
+
+        filenames = ['/dev/nvme0n1', '/dev/nvme1n1']
+        filename_option = '--filenames ' + ' '.join(filenames)
+
+        # Test case 2: turn off all tests
```

Comment:
maybe you should write different test cases in different `test_` functions?

Reply:
Fixed